music_assistant/providers/spotify/podcast_helpers.py

```diff
@@ -0,0 +1,485 @@
+            expiration=60 * 60 * 12,  # 12 hours for episodes (more dynamic content)
+        )
+
+    async def _cache_invalidate_podcast(self, podcast_id: str) -> None:
+        """Invalidate podcast and episodes cache."""
+        await self.mass.cache.delete(
+            key=f"{CACHE_KEY_PODCAST_PREFIX}{podcast_id}",
+            base_key=self.provider.lookup_key,
+            category=CACHE_CATEGORY_PODCASTS,
+        )
+        await self.mass.cache.delete(
+            key=f"{CACHE_KEY_EPISODES_PREFIX}{podcast_id}",
+            base_key=self.provider.lookup_key,
+            category=CACHE_CATEGORY_EPISODES,
+        )
+
+    # Core podcast methods
+    async def get_podcast(self, prov_podcast_id: str) -> Podcast:
+        """Get full podcast details by id with caching."""
+        # Try cache first
+        cached_data = await self._cache_get_podcast(prov_podcast_id)
+        if cached_data:
+            self.logger.debug(f"Using cached data for podcast {prov_podcast_id}")
+            return parse_podcast(cached_data, self.provider)
+
+        # Fetch from API if not cached
+        podcast_obj = await self.provider._get_data(f"shows/{prov_podcast_id}")
+
+        # Add a check to ensure podcast_obj is not None
+        if not podcast_obj:
+            raise ValueError(f"No podcast data returned from API for ID: {prov_podcast_id}")
+
+        await self._cache_set_podcast(prov_podcast_id, podcast_obj)
+        return parse_podcast(podcast_obj, self.provider)
+
+    async def get_podcast_episodes(
+        self, prov_podcast_id: str
+    ) -> AsyncGenerator[PodcastEpisode, None]:
+        """Get all podcast episodes with caching and played status sync to MA."""
+        # Try cache first
+        cached_episodes = await self._cache_get_episodes(prov_podcast_id)
+        if cached_episodes:
+            self.logger.debug(f"Using cached episodes for podcast {prov_podcast_id}")
+            podcast = await self.get_podcast(prov_podcast_id)
+
+            for episode_data in cached_episodes:
+                episode = parse_podcast_episode(episode_data, self.provider, podcast=podcast)
+                # Handle position assignment for cached episodes
+                if hasattr(episode, "_position"):
+                    episode.position = episode._position
+                yield episode
+            return
+
+        # Fetch fresh data from API
+        podcast = await self.get_podcast(prov_podcast_id)
+        episodes_to_cache: list[dict[str, Any]] = []
+        synced_count = 0
+        total_count = 0
+        episode_position = 1
+
+        # Paginate through all episodes from Spotify
+        page_size = 50
+        offset = 0
+
+        while True:
+            episodes_data = await self.provider._get_data(
+                f"shows/{prov_podcast_id}/episodes",
+                limit=page_size,
+                offset=offset,
+                market="from_token",  # Critical for getting user's resume_point data
+            )
```

Comment:
why not use the "get_all_items" helper ?

Reply:
The drawbacks are that get_all_items isn't currently setup to support caching like this one is. Also from the perspective of memory efficiency, get_all_items would require all episodes to be fetched into memory first which isn't ideal with a podcasts which can have hundreds of episodes.